pypeit/scripts/sensfunc.py

```diff
@@ -118,73 +120,156 @@ def main(args):
                        "              multi_spec_det = 3,7\n"
                        "\n")
 
-        # Determine the spectrograph and generate the primary FITS header
-        with io.fits_open(args.spec1dfile) as hdul:
-            spectrograph = load_spectrograph(hdul[0].header['PYP_SPEC'])
-            spectrograph_config_par = spectrograph.config_specific_par(hdul)
-
-            # Construct a primary FITS header that includes the spectrograph's
-            #   config keys for inclusion in the output sensfunc file
-            primary_hdr = io.initialize_header()
-            add_keys = (
-                ['PYP_SPEC', 'DATE-OBS', 'TELESCOP', 'INSTRUME', 'DETECTOR']
-                + spectrograph.configuration_keys() + spectrograph.raw_header_cards()
-            )
-            for key in add_keys:
-                if key.upper() in hdul[0].header.keys():
-                    primary_hdr[key.upper()] = hdul[0].header[key.upper()]
-
-        # If the .sens file was passed in read it and overwrite default parameters
-
-        if args.sens_file is not None:
-            sensFile = inputfiles.SensFile.from_file(args.sens_file)
-            # Read sens file
-            par = pypeitpar.PypeItPar.from_cfg_lines(cfg_lines=spectrograph_config_par.to_config(),
-                merge_with=(sensFile.cfg_lines,))
+
+        #Check if we want to compute a sensfile for every object in the directory
+        if args.dir:
+            msgs.info(f'Searching for files in directory: {args.dir}')
+            import glob
+            files_list = glob.glob(args.spec1dfile + 'spec1d*.fits')
+            print(files_list)
+            for file in files_list:
+                if ('spec1d' in file) and ('.fits' in file):
```

Comment:
There's a lot of repeated code below.  I would refactor so that the code always iterates through a list of spec1dfiles.  First, put `from pathlib import Path` with the rest of the imports at the beginning of the `main` function. Then here do:

```
if args.dir:
    files = sorted(Path(args.spec1dfile).resolve().glob('spec1d*.fits'))
    msgs.info(f'Spec 1D files found: {[f.name for f in files]})
else:
    files = [Path(args.spec1dfile).resolve()]

for file in files:
    if not file.exists():
        msgs.warn(f'{file} does not exist.')
        continue
    with io.fits_open(file) as hdul:
        ....
```
Then the rest continues as it did in the previous script.